cardano-testnet/src/Testnet/Start/Types.hs

```diff
@@ -34,37 +36,61 @@ import           Testnet.Filepath
+-- in the parser, and then get split into 'CardanoTestnetOptions' and
+-- 'ShelleyTestnetOptions'
+data CardanoTestnetCliOptions = CardanoTestnetCliOptions
+  { cliTestnetOptions :: CardanoTestnetOptions
+  , cliShelleyOptions :: ShelleyTestnetOptions
+  } deriving (Eq, Show)
+
+instance Default CardanoTestnetCliOptions where
+  def = CardanoTestnetCliOptions
+    { cliTestnetOptions = def
+    , cliShelleyOptions = def
+    }
 
+-- | Options which, contrary to 'ShelleyTestnetOptions' are not implemented
+-- by tuning the genesis files.
 data CardanoTestnetOptions = CardanoTestnetOptions
   { -- | List of node options. Each option will result in a single node being
     -- created.
     cardanoNodes :: [TestnetNodeOptions]
   , cardanoNodeEra :: AnyShelleyBasedEra -- ^ The era to start at
-  , cardanoEpochLength :: Int -- ^ An epoch's duration, in number of slots
-  , cardanoSlotLength :: Double -- ^ Slot length, in seconds
-  , cardanoTestnetMagic :: Int
-  , cardanoActiveSlotsCoeff :: Double
   , cardanoMaxSupply :: Word64 -- ^ The amount of Lovelace you are starting your testnet with (forwarded to shelley genesis)
+                               -- TODO move me to ShelleyTestnetOptions when https://github.com/IntersectMBO/cardano-cli/pull/874 makes it to cardano-node
   , cardanoEnableP2P :: Bool
   , cardanoNodeLoggingFormat :: NodeLoggingFormat
   , cardanoNumDReps :: Int -- ^ The number of DReps to generate at creation
   , cardanoEnableNewEpochStateLogging :: Bool -- ^ if epoch state logging is enabled
   } deriving (Eq, Show)
 
-cardanoDefaultTestnetOptions :: CardanoTestnetOptions
-cardanoDefaultTestnetOptions = CardanoTestnetOptions
-  { cardanoNodes = cardanoDefaultTestnetNodeOptions
-  , cardanoNodeEra = AnyShelleyBasedEra ShelleyBasedEraBabbage
-  , cardanoEpochLength = 500
-  , cardanoSlotLength = 0.1
-  , cardanoTestnetMagic = 42
-  , cardanoActiveSlotsCoeff = 0.05
-  , cardanoMaxSupply = 100_000_020_000_000 -- 100 000 billions Lovelace, so 100 millions ADA. This amount should be bigger than the 'byronTotalBalance' in Testnet.Start.Byron
-  , cardanoEnableP2P = False
-  , cardanoNodeLoggingFormat = NodeLoggingFormatAsJson
-  , cardanoNumDReps = 3
-  , cardanoEnableNewEpochStateLogging = True
-  }
+instance Default CardanoTestnetOptions where
+  def = CardanoTestnetOptions
+    { cardanoNodes = cardanoDefaultTestnetNodeOptions
+    , cardanoNodeEra = AnyShelleyBasedEra ShelleyBasedEraBabbage
+    , cardanoMaxSupply = 100_000_020_000_000 -- 100 000 billions Lovelace, so 100 millions ADA. This amount should be bigger than the 'byronTotalBalance' in Testnet.Start.Byron
+    , cardanoEnableP2P = False
+    , cardanoNodeLoggingFormat = NodeLoggingFormatAsJson
+    , cardanoNumDReps = 3
+    , cardanoEnableNewEpochStateLogging = True
+    }
+
+-- | Options that are implemented by writing fields in the Shelley genesis file.
+data ShelleyTestnetOptions = ShelleyTestnetOptions
```

Comment:
This split seems arbitrary. Why are we introducing another configuration type?

Reply:
I think the split is clear: `ShelleyTestnetOptions` contains values that are encoded in genesis files whereas `CardanoTestnetOptions` contains the rest of the values.

This is visible e.g. in the new prototype of [cardanoTestnet](https://github.com/IntersectMBO/cardano-node/blob/e6a9c24d44d3a567cdb415e1e28ab5e4791b2413/cardano-testnet/src/Testnet/Start/Cardano.hs#L188) that contains `CardanoTestnetOptions` and the genesis files, as opposed to [cardanoTestnetDefault](https://github.com/IntersectMBO/cardano-node/blob/e6a9c24d44d3a567cdb415e1e28ab5e4791b2413/cardano-testnet/src/Testnet/Start/Cardano.hs#L101). `cardanoTestnetDefault` has both `ShelleyTestnetOptions` and `CardanoTestnetOptions`, because the genesis files are not yet around at this stage.

Reply:
I think the source of confusion for me came from `cardanoMaxSupply` existing in both types. The naming is also confusing. Cardano vs Shelley Tesnetoptions. The difference should be immediately obvious in the names (you suggested `GenesisOptions` which is better).

We have:

```
data CardanoTestnetOptions = CardanoTestnetOptions
  { -- | List of node options. Each option will result in a single node being
    -- created.
    cardanoNodes :: [TestnetNodeOptions]
  , cardanoNodeEra :: AnyShelleyBasedEra -- ^ The era to start at
  , cardanoMaxSupply :: Word64 -- ^ The amount of Lovelace you are starting your testnet with (forwarded to shelley genesis)
                               -- TODO move me to ShelleyTestnetOptions when https://github.com/IntersectMBO/cardano-cli/pull/874 makes it to cardano-node
  , cardanoEnableP2P :: Bool
  , cardanoNodeLoggingFormat :: NodeLoggingFormat
  , cardanoNumDReps :: Int -- ^ The number of DReps to generate at creation
  , cardanoEnableNewEpochStateLogging :: Bool -- ^ if epoch state logging is enabled
  } deriving (Eq, Show)
```

I think we should aim for:
```
data CardanoTestnetOptions = CardanoTestnetOptions
  { cardanoNodes :: [TestnetNodeOptions]
  }
```

There are other random configuration values here e.g `cardanoEnableP2P` and `cardanoNumDReps` which do not belong here and then we have to reconcile configuration overrides which just leads to more confusion.

Reply:
These overrides can live in `create-testnet-data` but we also have the issue of relisting all of the potential configuration values in the `create-testnet-data` which obviously isn't sensible. Maybe we can move the existing overrides to `create-testnet-data`.